docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+runtime field and use values from that field at query time.
+
+Runtime fields are similar to the <<script-fields,`script_fields`>> parameter
+of the `_search` request, but also make the script results available for
+queries and aggregations.
+
+The script in the following request extracts the day of the week from the
+`@timestamp` field, which is defined as a `date` type:
+
+[source,console]
+----
+PUT /my-index
+{
+  "mappings": {
+    "runtime": { <1>
+      "day_of_week": {
+        "type": "keyword", <2>
+        "script": { <3>
+          "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+        }
+      }
+    },
+    "properties": {
+      "timestamp": {"type": "date"}
+    }
+  }
+}
+----
+
+<1> Runtime fields are defined in the `"runtime"` section of the mapping
+definition.
+<2> Each runtime has its own field type, just like any other field.
+<3> The script defines the evaluation to calculate at search time.
+
+The `"runtime"` section supports `boolean`, `date`, `double`, `ip`, `keyword`,
+and `long` data types. Runtime fields with a `type` of `date` can accept the
+<<mapping-date-format,`format`>> parameter exactly as the `date` field type.
+
+IMPORTANT: Objects are not supported in the `"runtime"` section under the
+mapping definition. If you want to map a runtime field under an object, you
+can
+<<runtime-shadowing-fields,shadow the field value in the search request>>.
+
+[[runtime-updating-scripts]]
+.Updating runtime scripts
+****
+
+Updating a script while a dependent query is running can return
+inconsistent results. Each shard might have access to different versions of the
+script, depending on when the mapping change takes effect.
+
+Existing queries or visualizations in {kib} that rely on runtime fields can
+fail if you change the field type. For example, a bar chart visualization
+that uses a runtime field of type `ip` will fail if the type is changed
+to `boolean`.
+
+****
+
+[discrete]
+[[runtime-search-request]]
+=== Defining runtime fields in a search request
+You can specify a `runtime_mappings` section in a search request to create
+runtime fields that exist only as part of the query. You specify a script
+as part of the `runtime_mappings` section, just as you would if adding a
+runtime field to the mappings.
+
+In the following request, the values for the `day_of_week` field are calculated
+dynamically, and only within the context of this search request:
+
+[source,console]
+----
+GET my-index/_search
+{
+  "runtime_mappings": {
+    "day_of_week": {
+      "type": "keyword",
+      "script": {
+        "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+      }
+    }
+  },
+  "aggs": {
+    "day_of_week": {
+      "terms": {
+        "field": "day_of_week"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+[discrete]
+[[runtime-shadowing-fields]]
+==== Shadowing fields in the search request
```

Comment:
why is this section specifically on shadowing fields in the search request? Shadowing can also be done when defining runtime fields in the mappings, so I would expect shadowing to be documented as a feature of runtime fields in general

Reply:
I've expanded the section on [shadowing fields](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime-shadowing-fields.html) to include both scenarios 👍